lib/fs/filesystem.go

```diff
@@ -260,6 +260,21 @@ func Canonicalize(file string) (string, error) {
 	return file, nil
 }
 
+// wrapFilesystem should always be used when wrapping a Filesystem.
+// It ensures proper wrapping order, which right now means:
+// `logFilesystem` needs to be the outermost wrapper for caller lookup.
```

Comment:
We have two different wrapping orders of logfs and walkfs, depending on whether we want to debug above or below walkfs... Probably that does not matter, and maybe we should remove the `walkfs` trace option as this is potentially confusing

Reply:
Having both `fs` and `walkfs` has merits: The former will just print one debug line that it will start walking now, the latter will also print every filesystem operation that occurs while walking - I think I remember I needed that huge walking verbosity once. And the unwrapping added in this PR is correct for both cases: It's appropriate for the filesystem calls originating in the walking code to give the fs walking code as the caller.